index.js

```diff
@@ -62,6 +62,32 @@ function Warehouse(options) {
   this.releaseLine = new Warehouse.ReleaseLine(this, options.releaseLine);
 }
 
+/**
+ * Normalize possible auth arguments
+ * @function _normalizeAuth
+ * @params {Object|String} opts Auth options from constructor
+ * @returns {String} to be used for authorization header
+ */
+Warehouse.prototype._normalizeAuth = function _normalizeAuth(opts) {
+  if (!opts) return;
+  // Legacy support is if opts is a string, we send it as a bearer token
+  if (typeof opts === 'string') return 'Bearer ' + opts;
+
+  if (!opts.type || !opts.token) return;
+
+  const type = opts.type.toLowerCase();
+
+  switch (type) {
+    case 'basic':
+      return `Basic ${opts.encoded ? opts.token : Buffer.from(opts.token, 'utf8').toString('base64')}`;
+    case 'bearer':
+      return `Bearer ${opts.token}`;
+    default:
+      return;
```

Comment:
should this eat the result and end up with no auth, or should it throw, warn, or whatever here?

Reply:
@msluther yea right now we are just silent and no auth, you'll get the error when an unearthed request is made

Reply:
we could throw

Reply:
I think in this case we should throw. they specified auth with a type, just that type isn't something we support

Reply:
rather than have a cryptic error of just auth didn't work (did I forget to supply it? did I get it wrong? did the credentials change? etc)